### ISSUE TITLE
Allow to use system installed QTermWidget and to conditionally build GitServerPlugin, JenkinsPlugin

### ISSUE THREAD

**Issue by band-a-prend** [francescmm/GitQlient#271]
**What current behavior doesn't fully fit your needs? Why?**
Currently the QTermWIdget plugin could be used as system installed `>=qtwemwidget-1.2.0` if user specify `Plugins Folder` as e.g. `/usr/lib64`. In this case user isn't allow to use `Download` of `Plugins` settings tab button to get GitServerPlugin, JenkinsPlugin.

**What you'd like to see? Propose a solution**
* Is it possible to use system installed QTermWIdget library if available in system with ignoring  `Plugins Folder` setting?
* Could the GitServerPlugin, JenkinsPlugin be shipped with GitQlient and conditionally built together with GitQlient as they all have the common dependencies `AuxiliarCustomWidgets`, `QLogger`, `git` submodules?

**Describe alternatives you've considered**
Alternatively please add this dependencies source fiules to it's tarballs too and use them as system installed if available instead of `Plugins Folder` setting.

**Comment by band-a-prend** [comment]
I realize that plugins codebase was extruded to external modules. I could build it as part of gitqlient by using package manager features but I have problems to patch gitqlient to point `/usr/lib64` etc as default PluginsFolder if `GitQlient.conf` were not generated previously. I.e. at first run for user.

**Comment by francescmm** [comment]
Hi @band-a-prend, originally the reason why GitQlient had it's own plugin is because qtermwidget was not available as plugin. It took a long time to have [the changes merged](https://github.com/lxqt/qtermwidget/pull/469). Now it should be possible with the right modifications on GitQlient.

The other plugins is something that I need to think about. I moved them out of GitQlient becuase it drag some dependencies I don't want to have in here. Specially the WebChannel/WebWidgets thing. Those are heavy Qt libraries that add quite a headache when having to create RPM/DEB packages.

**Comment by band-a-prend** [comment]
@francescmm 
Yes, it's ok. I remember that initially qtwebengine was gitqlient dependency when this plugins were added. And current splitting into application with external submodules is reasonable.

I successfully packaged the GitServerPlugin and JenkinsPlugin as separate packages with noticing user with post-install info how to setup PluginFolder path. The only current tiny problem is that used released tarballs of them contain empty "submodules" folders. Therefore I trick with replacing empty dirs with appropriate provided by GitQlient 1.6.1 release tarball. 

So providing submodules within plugins tarballs like it's done for GitQlient tarball will be nice.
